charts/datahub/templates/datahub-encryption-secrets.yml

```diff
@@ -1,11 +1,19 @@
 {{- $secret := lookup "v1" "Secret" .Release.Namespace "datahub-encryption-secrets" -}}
 {{- $data := $secret.data | default dict -}}
-{{- if .Values.global.datahub.encryptionKey.provisionSecret -}}
+{{- with .Values.global.datahub.encryptionKey.provisionSecret }}
+
+{{- if .enabled }}
 apiVersion: v1
 kind: Secret
 metadata:
   name: "datahub-encryption-secrets"
 type: Opaque
 data: 
-  encryption_key_secret: {{ index $data "encryption_key_secret" | default (randAlphaNum 20 | b64enc | quote) }}
-{{- end -}}
\ No newline at end of file
+  {{- if .autoGenerate }}
+  encryption_key_secret: {{ get $data "encryption_key_secret" | default (randAlphaNum 20 | b64enc | quote) }}
```

Comment:
does this mean we assume the user provides a base64 encoded secret?

Reply:
if yes can we mentioned that in the docs?

Reply:
No. If we set the provisioning secret logic to auto generate secrets it will:

1. Check if an existing secret with the same name already exists with: `$secret := lookup "v1" "Secret" .Release.Namespace "datahub-auth-secrets"`
2. Get the data entry with `get $data "encryption_key_secret"` (encoded), if none exists it will generate a random 20 character long alphanumerical string and base64 encode that.

User submitted secrets is the other part of ` {{- if .autoGenerate }}`: 
``` system_client_secret: {{ .secretValues.secret | b64enc | quote }}```